pkg/distributor/distributor.go

```diff
@@ -300,6 +305,31 @@ func (d *Distributor) Push(ctx context.Context, req *logproto.PushRequest) (*log
 	}
 }
 
+func (d *Distributor) truncateLines(vContext validationContext, stream *logproto.Stream) {
+	if !vContext.maxLineSizeTruncate {
+		return
+	}
+
+	var truncatedSamples, truncatedBytes int
+	for i, e := range stream.Entries {
+		if maxSize := vContext.maxLineSize; maxSize != 0 && len(e.Line) > maxSize {
+			indicator := vContext.maxLineSizeTruncateInd
+			truncateTo := maxSize - len(indicator)
+			if truncateTo <= 0 {
+				continue
+			}
+
+			stream.Entries[i].Line = e.Line[:truncateTo] + indicator
```

Comment:
Not a blocker, but I just realized this can split apart multi-byte runes (https://blog.golang.org/strings)